Controller/Webhook/Url.php

```diff
@@ -0,0 +1,163 @@
+use Magento\UrlRewrite\Model\UrlFinderInterface;
+use Magento\UrlRewrite\Model\UrlPersistInterface;
+use Magento\UrlRewrite\Model\UrlRewriteFactory;
+use Magento\UrlRewrite\Service\V1\Data\UrlRewrite;
+use Psr\Log\LoggerInterface;
+use stdClass;
+
+class Url implements HttpPostActionInterface, CsrfAwareActionInterface
+{
+    private RequestInterface $request;
+
+    private ConfigurationInterface $configuration;
+
+    private StoreManagerInterface $storeManager;
+
+    private Api $apiFactory;
+
+    private ResultFactory $resultFactory;
+
+    private UrlFinderInterface $urlFinder;
+
+    private UrlRewriteFactory $urlRewriteFactory;
+
+    private UrlPersistInterface $urlPersist;
+
+    private UrlRewriteResource $urlRewriteResource;
+
+    private LoggerInterface $logger;
+
+    public function __construct(
+        RequestInterface       $request,
+        ConfigurationInterface $configuration,
+        StoreManagerInterface  $storeManager,
+        Api                    $apiFactory,
+        ResultFactory          $resultFactory,
+        UrlFinderInterface     $urlFinder,
+        UrlRewriteFactory      $urlRewriteFactory,
+        UrlPersistInterface    $urlPersist,
+        UrlRewriteResource     $urlRewriteResource,
+        LoggerInterface        $logger
+    ) {
+        $this->request = $request;
+        $this->configuration = $configuration;
+        $this->storeManager = $storeManager;
+        $this->apiFactory = $apiFactory;
+        $this->resultFactory = $resultFactory;
+        $this->logger = $logger;
+        $this->urlFinder = $urlFinder;
+        $this->urlRewriteFactory = $urlRewriteFactory;
+        $this->urlPersist = $urlPersist;
+        $this->urlRewriteResource = $urlRewriteResource;
+    }
+
+    public function execute(): ResultInterface
+    {
+        $result = $this->resultFactory->create(ResultFactory::TYPE_JSON);
+
+        $payload = json_decode($this->request->getContent() ?? '', true);
+        if (!$payload) {
+            return $result->setData([
+                'success' => true
+            ]);
+        }
+
+        $store = $this->storeManager->getStore();
+        $urlRewriteDocumentTypes = $this->configuration->getUrlRewriteContentTypes(
+            $store
+        );
+
+        if (!$urlRewriteDocumentTypes) {
+            return $result->setData([
+                'success' => true
+            ]);
+        }
+
+        $documentIds = $payload['documents'] ?? [];
+        if (empty($documentIds)) {
+            return $result->setData([
+                'success' => true
+            ]);
+        }
+
+        $api = $this->apiFactory->create();
+        foreach ($documentIds as $documentId) {
+            $document = $api->getByID($documentId);
+
+            $urlRewrite = $this->findUrlRewrite($document, $store);
+
+            if ($urlRewrite && $urlRewrite->getEntityType() === CmsPageUrlRewriteGenerator::ENTITY_TYPE) {
```

Comment:
this adds a dependency on CmsPage, is this what we want? otherwise the composer.json and module.xml needs to be updated

Reply:
Yes, that's what we want. I've added them.